src/components/dialogs/stashed-elements/stashed-elements-dialog.tsx

```diff
@@ -0,0 +1,203 @@
+
+interface IStashedElementsDialog {
+    open: boolean;
+    onClose: () => void;
+}
+
+function getOptionLabel(element: any) {
+    return element.second
+        ? element.first.elementName + ' (' + element.second + ')'
+        : element.first.elementName;
+}
+
+function getElementId(element: any) {
+    return element.first.elementUuid;
+}
+
+const StashedElementsDialog: FunctionComponent<IStashedElementsDialog> = ({
+    open,
+    onClose,
+}) => {
+    const intl = useIntl();
+    const [selectedElements, setSelectedElements] = useState<string[]>([]);
+    const [elements, setElements] = useState<any[]>([]);
+    const [openConfirmationPopup, setOpenConfirmationPopup] =
+        useState<boolean>(false);
+    const { snackError } = useSnackMessage();
+
+    const selectedDirectory = useSelector(
+        (state: ReduxState) => state.selectedDirectory
+    );
+
+    const handleGetStashedElement = useCallback(() => {
+        getStashedElements()
+            .then((response: any[]) => {
+                setElements(response);
+            })
+            .catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                });
+            });
+    }, [snackError]);
+
+    useEffect(() => {
+        if (open) {
+            handleGetStashedElement();
+        }
+    }, [open, handleGetStashedElement]);
+
+    const handleSelectAll = useCallback(() => {
+        setSelectedElements((values) =>
+            values.length === elements.length
+                ? []
+                : elements.map((e) => getElementId(e))
+        );
+    }, [elements]);
+
+    const handleCheckBoxChange = useCallback((elementId: string) => {
+        setSelectedElements((values) =>
+            values.includes(elementId)
+                ? values.filter((id) => id !== elementId)
+                : [...values, elementId]
+        );
+    }, []);
+
+    const handleDelete = useCallback(() => {
+        deleteElements(selectedElements, selectedDirectory.elementUuid)
+            .then(() => handleGetStashedElement())
+            .catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                });
+            })
+            .finally(() => onClose());
+    }, [
+        selectedElements,
+        snackError,
+        selectedDirectory,
+        handleGetStashedElement,
+        onClose,
+    ]);
+
+    const handleRestore = useCallback(() => {
+        restoreElements(selectedElements, selectedDirectory.elementUuid)
+            .then(() => handleGetStashedElement())
+            .catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                });
+            })
+            .finally(() => onClose());
+    }, [
+        selectedElements,
+        snackError,
+        selectedDirectory,
+        handleGetStashedElement,
+        onClose,
+    ]);
+
+    const noSelectedElements = selectedElements.length === 0;
```

Comment:
Is this really useful?

Reply:
it used to disable 2 button so i created a constant